flang/include/flang/Tools/CLOptions.inc

```diff
@@ -70,40 +70,16 @@ inline void addCSE(mlir::PassManager &pm) {
       pm, disableFirCse, fir::createCSEPass);
 }
 
-#if !defined(FLANG_EXCLUDE_CODEGEN)
-inline void addCodeGenRewritePass(mlir::PassManager &pm) {
-  addPassConditionally(
-      pm, disableCodeGenRewrite, fir::createFirCodeGenRewritePass);
-}
-
-inline void addTargetRewritePass(mlir::PassManager &pm) {
-  addPassConditionally(pm, disableTargetRewrite, []() {
-    return fir::createFirTargetRewritePass(fir::TargetRewriteOptions{});
-  });
-}
-
-inline void addFIRToLLVMPass(mlir::PassManager &pm) {
-  addPassConditionally(pm, disableFirToLlvmIr, fir::createFIRToLLVMPass);
-}
-
-inline void addLLVMDialectToLLVMPass(
-    mlir::PassManager &pm, llvm::raw_ostream &output) {
-  addPassConditionally(pm, disableLlvmIrToLlvm,
-      [&]() { return fir::createLLVMDialectToLLVMPass(output); });
-}
-
-/// Create a pass pipeline for lowering from MLIR to LLVM IR
+/// Create a basic pass pipeline for simplyfing the FIR dialect
 ///
 /// \param pm - MLIR pass manager that will hold the pipeline definition
-inline void createMLIRToLLVMPassPipeline(mlir::PassManager &pm)
+inline void createBasicFirPassPipeline(mlir::PassManager &pm)
```

Comment:
This name change obfuscates what this pass pipeline actually does and effectively reverts a name change request made during upstreaming efforts. The pipeline takes MLIR and converts it to LLVM.

Reply:
Only `createBasicFirPassPipeline` is moved. That's required in order for `bbc` to be able to use it. Methods for code-gen are [excluded](https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/tools/bbc/bbc.cpp#L133-L134) in `bbc`. However, with this change `createBasicFirPassPipeline` becomes a method that we do want to use in `bbc` (I removed code-gen passes from there).

The diff suggests that other methods are moved as well, but that's how `git diff` is displayed. There isn't much I can do about this.

Reply:
> This name change obfuscates what this pass pipeline actually does

Please, could you suggest a more suitable change?

> effectively reverts a name change request made during upstreaming efforts

I don't believe that's the case. I introduced this method in https://github.com/flang-compiler/f18-llvm-project/pull/1113 to share code with `tco` and `flang-new`. Now it seems like a good time to refactor it so that it can be shared with `bbc` too. Also, AFAIK, this method is not present upstream _yet_?